server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -154,18 +155,23 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
 
     @Override
     public Mono<byte[]> readBytes(BucketName bucketName, BlobId blobId) {
-        return Mono.just(bucketName)
-            .filter(getDefaultBucketName()::equals)
-            .flatMap(deleteBucket -> readBytesInDefaultBucket(bucketName, blobId))
-            .switchIfEmpty(readBytesFromBackend(bucketName, blobId));
+        if (getDefaultBucketName().equals(bucketName)) {
+            return readBytesInDefaultBucket(bucketName, blobId);
+        }
+        return readBytesFromBackend(bucketName, blobId);
```

Comment:
:+1:

Reply:
Also a bug here, with the old code, when we read bytes does not exist:
readFromCache => readFromBackend => switchEmpty(readFromBackend) again.

Reply:
(can you write a test about it then?)

Reply:
It is already in our test siute